File: frontend/src/types/timelinePremium.ts
import type { ChecklistItemPremium } from "./checklistPremium";

export interface TimelinePremiumSetRequest {
  session_id: string;
  wedding_date: string;
}

export interface TimelinePremiumResponse {
  session_id: string;
  wedding_date: string | null;
  items: ChecklistItemPremium[];
}

File: backend/app/schemas/timelinePremium.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

from app.schemas.checklistPremium import ChecklistItemPremiumSchema


class TimelinePremiumSetRequest(BaseModel):
    session_id: str
    wedding_date: str  # format ISO: "2027-06-20"


class TimelinePremiumResponse(BaseModel):
    session_id: str
    wedding_date: Optional[datetime]
    items: List[ChecklistItemPremiumSchema]